clang/lib/CodeGen/Targets/RISCV.cpp

```diff
@@ -251,6 +253,15 @@ bool RISCVABIInfo::detectFPCCEligibleStructHelper(QualType Ty, CharUnits CurOff,
         // bitwidth is XLen or less.
         if (getContext().getTypeSize(QTy) > XLen && BitWidth <= XLen)
           QTy = getContext().getIntTypeForBitwidth(XLen, false);
+        // Trim type to alignment/bitwidth if that is possible
+        else if (getContext().getTypeSize(QTy) > Alignment &&
```

Comment:
Is the alignment relevant? Should we just round the bitfield size to the smallest power of 2 always? Even without the packed attribute, gcc -O0 emits `sb` while clang emits `sw`.

Reply:
You are right, the latest push reflects the change suggested. It now just checks against size of a byte to ensure the size at the very minimum is 8 bits.